lib/msal-native-auth/src/controller/NativeAuthStandardController.ts

```diff
@@ -0,0 +1,229 @@
+import { SignInResult } from "../auth_flow/result/SignInResult.js";
+import { SignUpResult } from "../auth_flow/result/SignUpResult.js";
+import { InvalidArgumentError } from "../error/InvalidArgumentError.js";
+import { UnexpectedError } from "../error/UnexpectedError.js";
+import {
+    SignInStartParams,
+    SignInSubmitPasswordParams,
+} from "../interaction_client/parameter/SignInParams.js";
+import { SignInWithContinuationTokenResult } from "../interaction_client/result/SignInActionResult.js";
+import { SigninClient } from "../interaction_client/SignInClient.js";
+import {
+    GetAccountInputs,
+    SignInInputs,
+    SignUpInputs,
+    ResetPasswordInputs,
+    NativeAuthActionInputs,
+} from "../NativeAuthActionInputs.js";
+import { NativeAuthConfiguration } from "../NativeAuthConfiguration.js";
+import { NativeAuthApiClient } from "../network_client/NativeAuthApiClient.js";
+import { SignInCodeSendResponse } from "../network_client/response/SignInResponse.js";
+import { NativeAuthOperatingContext } from "../operating_context/NativeAuthOperatingContext.js";
+import { INativeAuthStardardController } from "./INativeAuthStandardController.js";
+
+/*
+ * Controller for standard native auth operations.
+ */
+export class NativeAuthStandardController
+    extends StandardController
+    implements INativeAuthStardardController
+{
+    /*
+     * The client to use for sign-in operations.
+     */
+    private readonly signInClient: SigninClient; // More clients will be added for sign-up, reset password, etc.
+
+    /*
+     * The configuration for the client.
+     */
+    private readonly nativeAuthConfig: NativeAuthConfiguration;
+
+    /*
+     * Constructor for NativeAuthStandardController.
+     * @param operatingContext - The operating context for the controller.
+     */
+    constructor(operatingContext: NativeAuthOperatingContext) {
+        super(operatingContext);
+
+        this.nativeAuthConfig = operatingContext.getNativeAuthConfig();
+
+        const nativeAuthApiClient = new NativeAuthApiClient(new FetchClient());
+        this.signInClient = new SigninClient(nativeAuthApiClient);
+        // Create more interaction clients here, such as SignUpClient, ResetPasswordClient, etc.
+    }
+
+    /*
+     * Gets the current account from the cache.
+     * @param getAccountOptions - Options for getting the current cached account
+     * @returns - A promise that resolves to GetAccountResult
+     */
+    async getCurrentAccount(
+        getAccountOptions: GetAccountInputs
+    ): Promise<GetAccountResult> {
+        const correlationId = this.getCorrelationId(getAccountOptions);
+
+        throw new Error(
+            `Method not implemented with Parameter ${correlationId}.`
+        );
+    }
+
+    /*
+     * Signs the user in.
+     * @param signInOptions - Options for signing in the user.
+     * @returns The result of the operation.
+     */
+    async signIn(signInOptions: SignInInputs): Promise<SignInResult> {
```

Comment:
as discussed, IMHO, this call should return the `signInClient` instead of a signInResult, or eventually allow the user to specify the flow

Reply:
I don't think an action call should return any *Client type. Let us discuss this in the review meeting if you still have strong opinion for this. Thanks.